src/components/Board.tsx

```diff
@@ -1,15 +1,100 @@
 import React from "react";
 import { useContext } from "react";
 import { GameContext } from "../context/GameContext";
-import { startGame } from "../logic/actions";
+import { startGame, discoverCell, markCell } from "../logic/actions";
+
+import styled from "styled-components";
+import { GridElement, GridElementState } from "../logic/board";
+import { GameStatus } from "../logic/reducer";
+import PropTypes from "prop-types";
+
+const Grid = styled.div`
+  display: grid;
+  grid-template-columns: repeat(${props => props.sizeX}, 2rem);
+  grid-template-rows: repeat(${props => props.sizeY}, 2rem);
+
+  .cell {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    &.mine {
+      background: red;
+    }
+
+    &.marked {
+      background: blue;
+    }
+
+    &.unsure {
+      background: grey;
+    }
+
+    &.failure {
+      background: green;
+    }
+
+    &.hidden {
+      background: #ccc;
+      border: 3px inset #ccc;
+    }
+  }
+`;
+
+interface CellProps {
+  cell: GridElement;
+}
+
+const Cell: React.FC<CellProps> = ({ cell }) => {
+  const { dispatch } = useContext(GameContext);
+  const classMapping = {
+    [GridElementState.Hidden]: "hidden",
+    [GridElementState.MarkedAsMine]: "marked",
```

Comment:
Just to make sure `MarkedAsMine` means the flag right?

Reply:
Well in the Windows version it was a flag, yes :)